### PR TITLE
Fix positive and negative offsets causing Plug-Ins to run twice

### LINKED ISSUE #127: Using positive and negative offsets leading to some Plug-Ins being processed multiple times
When using a mix of Plug-Ins with positive and negative offsets, some Plug-Ins get executed twice, depending on where you continue the publishing process.

Example structure:

- `Collector1` (order=CollectorOrder)
- `Collector2` (order=CollectorOrder + 0.4)
- `Validator1` (order=ValidatorOrder - 0.4)
- `Validator2` (order=ValidatorOrder)
- `Validator3` (order=ValidatorOrder + 0.4)
- `Extractor1` (order=ExtractorOrder - 0.4)
- `Extractor2` (order=ExtractorOrder)
- `Integrator`

With these example Plug-Ins, we can run into either `Collector2` or `Validator3` being run twice. Here's how:

- Open Pyblish Lite UI
- Collection will run
- Pressing the "Play" Button now will lead to `Collector2` being processed an additional time.
- Instead, if we press the "Validate" Button first, all Validators run as expected without any Collectors being run twice.
- If we now press the "Play" Button, `Validator3` will be processed an additional time.

This problem seems to happen [here](https://github.com/pyblish/pyblish-lite/blob/d25fe01b894a29285f7d6e64e6362caee45e8361/pyblish_lite/control.py#L95), because we take the `.order` of the next Plug-In to run and **always** subtract 0.5 from it to find out where we should start. This in turn causes an issue if the next order is actually using a negative offset and will go back more than it needs to:

- Start of Collection
- `Collector1` is processed
- `Collector2` is processed (`0 + 0.4 = 0.4`)
- Press "Publish" button
- Next Plug-In is `Validator1` (`1 - 0.4 = 0.6`)
- We subtract 0.5 from this to figure out where to continue (`0.6 - 0.5 = 0.1`)
- `Collector2` is processed **again**
- `Validator1` is processed
- `Validator2` is processed
- ...

### PR DESCRIPTION
**Pull request by Etherkirby**
Closes #127 

@BigRoy Suggested to call `round` on the `plugin.order` before we subtract 0.5 from it, but after some testing I found this to be a bit unsafe because of rounding errors (`round(0.5)` --> `0`).

### CODE REVIEW

File: pyblish_lite/control.py [hunk 1]
```diff
@@ -240,16 +240,16 @@ def on_finished_():
...
         self.is_running = True
         self.pair_generator = self._iterator(
             self.plugins,
             self.context,
-                                             
-            # Minus 0.5, because each order is a range of values,
-            # from -0.5 to 0.5. E.g. ExtractorOrder is 2, and spans
-            # between 1.5-2.5
-            start_from - 0.5
+            start_from
```
[the remark this answers is not in the document]
Comment (Etherkirby) [reply pyblish_lite/control.py:252]: Weirdly enough, at the point where `publish` is getting the "current pair", `self.current_pair[0]` is actually pointing to the **next** Plug-In to evaluate: https://github.com/pyblish/pyblish-lite/blob/6f68284d0a64b5d2f0c65de21c22a1bedb12638c/pyblish_lite/control.py#L91

Here's a snippet that proves this:

```python
from __future__ import print_function

import os

# Remove artificial delay from GUI
os.environ["PYBLISH_DELAY"] = "0"

import pyblish.api
import pyblish.lib
from pyblish_lite import control

count = {"#": 0}

class MyCollector(pyblish.api.ContextPlugin):
    order = pyblish.api.CollectorOrder

    def process(self, context):
        count["#"] += 1

class MyCollectorPosOffset(pyblish.api.ContextPlugin):
    order = pyblish.api.CollectorOrder + 0.4

    def process(self, context):
        count["#"] += 1

class MyValidatorNegOffset(pyblish.api.ContextPlugin):
    order = pyblish.api.ValidatorOrder - 0.4

    def process(self, context):
        count["#"] += 10

class MyValidator(pyblish.api.ContextPlugin):
    order = pyblish.api.ValidatorOrder

    def process(self, context):
        count["#"] += 10

pyblish.api.deregister_all_plugins()

test_plugins = (
    MyCollector, MyCollectorPosOffset, MyValidatorNegOffset, 
    MyValidator
)
for plugin in test_plugins:
    pyblish.api.register_plugin(plugin)

ctrl = control.Controller()
print("Current pair before collection:", ctrl.current_pair)
ctrl.reset()
print("Current pair after collection:", ctrl.current_pair)
assert count["#"] == 2, count

ctrl.publish()
print("Current pair after publish:", ctrl.current_pair)
# Disabling cause this fails in the current version and is getting fixed here.
# assert count["#"] == 22, count
```

-->

```
Current pair before collection: (None, None)
Current pair after collection: (<class 'pyblish.plugin.MyValidatorNegOffset'>, None)
Current pair after publish: (None, None)
```

TL;DR: It's actually using the order of the next Plug-In which wasn't processed yet, so no Plug-Ins are processed twice.

**Comment by mottosso** [comment]
Closing based on discussion in #127
